src/Components/OATPropertyEditor/Internal/FormRootModelDetails/Internal/FormRootModelDetailsContent/PropertyDetailsEditorModalContent.tsx

```diff
@@ -0,0 +1,731 @@
+    const [hasDescriptionError, setHasDescriptionError] = useState(null);
+    const [displayNameError, setHasDisplayNameError] = useState(null);
+
+    // callbacks
+
+    const onValidateNumber = useCallback(
+        (currentValue: string, newValue: string) => {
+            if (!isDefined(newValue) || newValue.trim() === '') {
+                return null;
+            }
+
+            const number = Number(newValue.trim());
+            if (!isNaN(number)) {
+                return newValue;
+            }
+            return currentValue;
+        },
+        []
+    );
+
+    // data
+    const displayNameMultiLangOptions: IChoiceGroupOption[] = [
+        {
+            key: SINGLE_LANGUAGE_KEY,
+            text: t('OATPropertyEditor.singleLanguage'),
+            disabled: multiLanguageDisplayNames.length > 0
+        },
+        {
+            key: MULTI_LANGUAGE_KEY,
+            text: t('OATPropertyEditor.multiLanguage')
+        }
+    ];
+
+    const descriptionMultiLangOptions: IChoiceGroupOption[] = [
+        {
+            key: SINGLE_LANGUAGE_KEY,
+            text: t('OATPropertyEditor.singleLanguage'),
+            disabled: multiLanguageDescriptions.length > 0
+        },
+        {
+            key: MULTI_LANGUAGE_KEY,
+            text: t('OATPropertyEditor.multiLanguage')
+        }
+    ];
+
+    // side effects
+    // Update multiLanguageSelectionsDisplayNames on every new language change
+    useEffect(() => {
+        if (typeof selectedItem.displayName === 'object') {
+            // Check if array of object includes empty values
+            const hasEmptyValues = multiLanguageDisplayNames.some(
+                (language) => selectedItem.displayName[language] === ''
+            );
+            setIsAMultiLanguageDisplayNameEmpty(hasEmptyValues);
+        }
+    }, [multiLanguageDisplayNames, selectedItem.displayName]);
+
+    // Update multiLanguageSelectionsDescriptions on every new language change
+    useEffect(() => {
+        if (typeof selectedItem.description === 'object') {
+            // Check if array of object includes empty values
+            const hasEmptyValues = multiLanguageDescriptions.some(
+                (language) => selectedItem.description[language] === ''
+            );
+            setIsAMultiLanguageDescriptionEmpty(hasEmptyValues);
+        }
+    }, [multiLanguageDescriptions, selectedItem.description]);
+
+    // styles
+    const propertyInspectorStyles = getPropertyInspectorStyles();
+    const radioGroupRowStyle = getRadioGroupRowStyles();
+    const classNames = getClassNames(styles, {
+        theme: useExtendedTheme()
+    });
+
+    logDebugConsole('debug', 'Render. {selectedItem}', selectedItem);
+
+    return (
+        <div className={classNames.root}>
+            {/* display name */}
+            <div className={propertyInspectorStyles.modalRow}>
+                <Label className={classNames.label} id={'display-name-label'}>
```

Comment:
Id and AriaLabelledBy don't need to be in `{}` since they aren't variabled. could be `id="display-name-label">`